nvflare/private/fed/server/job_runner.py

```diff
@@ -77,18 +77,19 @@ def _deploy_job(self, job: Job, sites: dict, fl_ctx: FLContext) -> str:
                         fl_ctx, f"Application {app_name} deployed to the server for run:{run_number}", fire_event=False
                     )
                     if not success:
-                        raise RuntimeError("Failed to deploy the App to the server")
+                        raise RuntimeError(f"Failed to deploy the App: {app_name} to the server")
```

Comment:
is it possible to tell user why the deployment failed, networking, configuration, etc. etc. 
instead generic deployment failed ?  

Some instructive message, so user knows what failed, and he has next thing to do: fix network, start server, change configuration, fix code etc.

Reply:
How the app deployment works is loading the App data from the Job, and sending thee data to the client to unzip to the workspace. At this point, the Job has passed the validation, the job data structure and configuration meta.json already been validated. So the deployment failure only means the App failed to be sent to the client.